### ISSUE TITLE
` consistent-boolean-name`: add more options for prefixes, add ignore patterns, pluralization

### ISSUE THREAD

**Issue by seofl** [sindresorhus/eslint-plugin-unicorn#3410]
### Description

Currently it feels like consistent boolean name is too restrictive.

Adding the ability to add own prefixes or/and add tokens to be ignored would allow users to have more control over it.
Either way,  this rule doesn't account for words that are in plural form, things like `areFilesValid` should pass 

### Examples

```ts
// example config
{ 
  rules: {
     "unicorn/consistent-boolean-name": [
        "error", 
        { 
            ignorePatterns: ["value"],
            prefixes: {
                are: true
            }
        }
     ]
}
```
```ts
// ✅
const emit = defineEmits<{
	(e: "update:intermidiate", value: boolean): void;
	(e: "value-change", value: boolean): void;
	(e: "change", event: Event): void;
}>();
```
```ts
// ❌
function validateFiles(fileList: FileList) : boolean {
// ...
}

// ✅ - currently fails
function areFilesValid(fileList: FileList) : boolean {
// ...
}
```
